src/waveform/waveformwidgetfactory.cpp

```diff
@@ -432,6 +441,22 @@ void WaveformWidgetFactory::setZoomSync(bool sync) {
     }
 }
 
+void WaveformWidgetFactory::setDisplayBeatGrid(bool sync) {
+    m_beatGridEnabled = sync;
+    if (m_config) {
+        m_config->set(ConfigKey("[Waveform]", "beatGridLinesCheckBox"), ConfigValue(m_beatGridEnabled));
```

Comment:
we normally do not set the value, if not required. We try to load the value and if it fails pick the default. Is it required here?